### DISCUSSION TITLE
Additional larger scaling options for calendar

### DISCUSSION THREAD

**Discussion by Ariandr** [pakerwreah/Calendr#244] [Ideas]
Hi.
Thank you for the very useful tool, appreciate your work.
But I have a feature request.

My eyesight is not perfect and I would love to have bigger scaling options, so that I could see the calendar dates without squeezing my eyes. It would be nice having extra large options, so that everything is bigger, especially the dates itself.

I set the calendar to the biggest scale, but still I struggle a lot, cannot see it at a glance as in larger calendar applications.
![image](https://github.com/user-attachments/assets/b8e28d24-6faa-4711-a69a-998fe5744dd9)

Thank you a lot.

**Comment by pakerwreah** [comment]
[Calendr.zip](https://github.com/user-attachments/files/17007816/Calendr.zip)

There you go, hope that's enough :)

**Comment by Ariandr** [comment]
Hi @pakerwreah 
Yep, that's way better. Thank you a lot!
I set the biggest option now and it's comfortable for me.

I think having more options for all users won't hurt anyone.
Appreciate it!

**Comment by Ariandr** [comment]
@pakerwreah 
I saw the code you changed, one line. Well though through🙂

But now I'm thinking, maybe it's even better to make 1.5 as the final scaling destination with 2-3 additional scaling options?
I could definitely use even bigger dates and it gives more width real estate on the bottom for all events content.

But of course don't want to be intrusive if you find it too much.
Thank you in any case!

**Comment by pakerwreah** [comment]
Well, yeah, that gets quite disproportionate to the rest of the UI.

I'm wondering how can you even read the events.

**Comment by Ariandr** [comment]
@pakerwreah 
Well, probably in the future it would be good to have scaling for text fonts too. But I usually see the color of the calendar events and open the full calendar app.

I use the widget calendar mainly to glance through the dates and see if I have anything scheduled, but for details usually open the full calendar.

But the calendar dates, as for me, don't have to scale together with the rest of the app, because when I had better eyesight 5 years ago I still used the app with big calendar dates and it was comfortable.

Still have this app installed (Calendars), but it haven't been updated in 5 years and I had constant issues with it.
![image](https://github.com/user-attachments/assets/c47cc4f4-3573-4873-98b2-ce172b526045)

**Comment by Ariandr** [comment]
That's regarding proportions, when both opened at the same time.
![image](https://github.com/user-attachments/assets/edf644ae-96aa-48b0-95e2-3671cc0f56a0)

**Comment by pakerwreah** [comment]
That makes sense.
It would be very hard for me to scale the entire UI dynamically.
I will see if I can do something about it as a separate accessibility feature.
That way I don't have to worry about breaking the entire app when it's enabled 😅.
As long as it gets good enough for you to read I will consider it a win.

In the meanwhile, I guess you could try lowering your screen resolution? That way the whole system gets bigger. (you probably already did it)

**Comment by Ariandr** [comment]
@pakerwreah  
Yes, I already have bigger scaling of the screen than most people, I think😅

For me the app got way better with the 1.4 scaling option, now I can relatively comfortably use it as a glanceable full calendar substitution to view the dates and do quick calculations in my mind. Having it more than 1.4 would be nice as long as it doesn't break anything, maybe people as me can benefit from it. But it's optional.

Thank you for your considerations.

**Comment by pakerwreah** [comment]
@Ariandr hey! Check out the latest version. I think you will like it 😉

**Comment by Ariandr** [comment]
@pakerwreah 
Great, thank you very much!
Accessibility options are awesome🔥🔥🔥

**Comment by Ariandr** [comment]
@pakerwreah 
Really awesome, you did a great job❤️
The ultimate improvement I can suggest for accessibility is adding a switch to not increase Calendar dates fonts. The ones that are already scaled up by the setting to scale the calendar.

Essentially, when enabled, it would increase all the fonts as now except the fonts for dates, so those would stay as if without accessibility turned up.
![image](https://github.com/user-attachments/assets/2e565e68-1b6e-4a05-abae-41ec333edc42)

I scaled the calendar to the largest option (it's just amazing, by the way❤️) and when I turn up accessibility, the dates become too condensed when the font increases, I don't quite need this huge font for dates specifically, since they are already big🙂

**Comment by pakerwreah** [comment]
Interesting, my thought was exactly the opposite 🤔 
I thought people would want it to increase the dates so it would be easier to read.

**Comment by pakerwreah** [comment]
If you don't go until the end of the slider it looks good.
I think it's a good compromise, because I'm pretty sure someone will ask me to scale the dates a well.
I'm gonna promote this to a Discussion so I can close the issue.

**Comment by Ariandr** [comment]
@pakerwreah 
The switch I suggest would be optional, so like additional functionality. People can use it or leave it as it is in the current version.
By default it would be turned off and everything would be like it is now in the current version.

But if you enable it, the date fonts would not increase based on Accessibility. So, when the switch is enabled, the accessibility setting would not affect the fonts on the calendar UI (that scales from the Calendar scale option).

**Comment by Ariandr** [comment]
@pakerwreah 
Actually, I came up with this idea. What do you think?

https://github.com/user-attachments/assets/cf08bab8-f5b8-42b2-98b7-8738d5ee8dfd

**Reply by pakerwreah** [reply]
That looks great! 👏🏻 

I think we can move the calendar resizing to this tab as well and just call that section "Calendar", so we can add more stuff there in the future that are not necessarily accessibility related. We can reuse `"settings.calendar"` localization for that.

Are you preparing a PR or should I work on that?

**Reply by Ariandr** [reply]
@pakerwreah 
Great, happy to hear that👍
I will prepare and submit PR, yes.
Probably you can move calendar scaling to this tab after my changes.

**Reply by pakerwreah** [reply]
Awesome! Go for it! 🚀

**Comment by Ariandr** [comment]
@pakerwreah 
Done. Please have a look: https://github.com/pakerwreah/Calendr/pull/246
